src/test/java/org/scijava/module/ModuleServiceTest.java

```diff
@@ -158,6 +159,66 @@ public void testGetSingleInput() throws ModuleException {
 		assertSame(info.getInput("double2"), singleDouble);
 	}
 
+	@Test
+	public void testSaveAndLoad() {
+
+		List<Object[]> parameterizations = new ArrayList<>();
+		parameterizations.add(new Object[] { //
+			double[].class, new double[] {} //
+		});
+		parameterizations.add(new Object[] { //
+			double[].class, new double[] { 1., 2., 3. } //
+		});
+		parameterizations.add(new Object[] { //
+			Double[].class, new Double[] {} //
+		});
+		parameterizations.add(new Object[] { //
+			Double[].class, new Double[] { 1., 2., 3. } //
+		});
+
+		for (Object[] params : parameterizations) {
+			Class<?> type = (Class<?>) params[0];
+			Object expected = params[1];
+			// Get a ModuleItem of the right type
+			MutableModule m = new DefaultMutableModule();
+			m.getInfo().addInput(new DefaultMutableModuleItem<>(m, "a", type));
+			@SuppressWarnings("unchecked")
+			final ModuleItem<Object> item = (ModuleItem<Object>) moduleService
```

Comment:
Casting to `ModuleItem<Object>` here is wrong. It must be `ModuleItem<?>` because we don't actually know the module item's type. The type is not actually `Object` in any of the four concrete cases declared above.

I understand that this was done so that the following `save` call would compile, but the correct way to do this would be to have a private helper function parameterized on `T` that accepts the `expected` as a `T` argument, so that you have a type variable to declare the return value of `getSingleInput` as a `ModuleItem<T>`, and then subsequently use it with `save`.

Reply:
Thanks, I fixed it in the most recent commit.